### PR TITLE
Added an ability to specify cluster secret in replicated database

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
Changelog category (leave one):

- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added an ability to specify cluster secret in replicated database.

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
Need to mark test `01034_move_partition_from_table_zookeeper` with `no-backward-compatibility-check` tag

File: src/Databases/DatabaseReplicatedSettings.h [hunk 1]
```diff
@@ -7,13 +7,24 @@ namespace DB
...
+    M(String, collection_name, "", "A name of a collection defined in server's config where all info for cluster authentication is defined", 0) \
+
+#define MAKE_OBSOLETE_REPLICATED_DATABASE_SETTING(M, TYPE, NAME, DEFAULT) \
+    M(TYPE, NAME, DEFAULT, "Obsolete setting, does nothing.", BaseSettingsHelpers::Flags::OBSOLETE)
+
+#define OBSOLETE_DATABASE_REPLICATED_SETTINGS(M) \
+    /** Obsolete settings that do nothing but left for compatibility reasons. Remove each one after half a year of obsolescence. */ \
+    MAKE_OBSOLETE_REPLICATED_DATABASE_SETTING(M, String, cluster_username, "") \
+    MAKE_OBSOLETE_REPLICATED_DATABASE_SETTING(M, String, cluster_password, "") \
+    MAKE_OBSOLETE_REPLICATED_DATABASE_SETTING(M, Bool, cluster_secure_connection, false) \
```
Comment (tavplubix) [inline src/Databases/DatabaseReplicatedSettings.h:23]: The feature is still experimental, so we can simply remove these settings.

File: src/Databases/DatabaseReplicatedSettings.h [hunk 2]
```diff
@@ -7,13 +7,24 @@ namespace DB
 
 class ASTStorage;
 
-#define LIST_OF_DATABASE_REPLICATED_SETTINGS(M) \
-    M(Float, max_broken_tables_ratio, 0.5, "Do not recover replica automatically if the ratio of staled tables to all tables is greater", 0) \
+#define CONTEMPORARY_DATABASE_REPLICATED_SETTINGS(M) \
+    M(Float,  max_broken_tables_ratio, 0.5, "Do not recover replica automatically if the ratio of staled tables to all tables is greater", 0) \
```
Comment (tavplubix) [inline src/Databases/DatabaseReplicatedSettings.h:11]: double space

File: src/Databases/DatabaseReplicatedSettings.h (hunk 1 above)
Comment (tavplubix) [reply src/Databases/DatabaseReplicatedSettings.h:23]: Yep, it will not work anyway, because the settings are not used now, so let's remove

**Review by tavplubix — APPROVED** [review]

File: src/Databases/DatabaseReplicatedSettings.h (hunk 2 above)
Comment (nikitamikhaylov) [reply src/Databases/DatabaseReplicatedSettings.h:11]: This is intended to make the code look nice 
image

File: src/Databases/DatabaseReplicatedSettings.h (hunk 1 above)
Comment (nikitamikhaylov) [reply src/Databases/DatabaseReplicatedSettings.h:23]: Ok.

File: src/Databases/DatabaseReplicatedSettings.h (hunk 2 above)
Comment (tavplubix) [reply src/Databases/DatabaseReplicatedSettings.h:11]: We don't have such indentation in Settings.h or MergeTreeSettings.h, but ok

**Comment by nikitamikhaylov** [comment]
Failures are irrelevant.

**Pull request merged**